### PR TITLE
Review: Integration tests and SDK fixes

### PR DESCRIPTION
**Pull request by swesam**
## For Copilot Review

This PR shows the full diff of integration test additions and SDK fixes
merged in #45, for a final review pass.

## Changes

- Fix v3 delete methods crashing on 204 No Content (centralized in requestV3)
- Fix getAnalytics crash on missing params (union type + runtime validation)
- Add comprehensive integration test suite (44 tests, gated behind opt-in flag)
- Add unit tests for 204 handling and analytics edge cases

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -2076,12 +2080,32 @@ export class RuleClient {
...
-    searchParams.set('object_type', params.object_type);
-    for (const id of params.object_ids) {
-      searchParams.append('object_ids[]', id);
+    if (!('object_type' in params)) {
+      const p = params as unknown as Record<string, unknown>;
+      if (p.object_ids != null || p.metrics != null) {
+        throw new RuleConfigError(
+          'object_ids and metrics require object_type to be provided'
+        );
+      }
```
[the remark this answers is not in the document]
Comment (swesam) [reply src/client.ts]: Fixed — `hasObjectType` now checks both key presence and truthiness, so `object_type: undefined` from JS spreads triggers the orphaned-params error. Added unit test for this case. Fixed in fb72a0d.

File: tests/integration.test.ts [hunk 2]
```diff
@@ -0,0 +1,607 @@
...
+      // Register cleanup so the subscriber is removed even if later tests fail
+      cleanup.push(async () => {
+        await client.deleteSubscriber(testEmail);
+      });
+    });
+
+    it('should retrieve subscriber fields', async () => {
+      // Small delay to let Rule.io propagate
+      await new Promise((r) => setTimeout(r, 2000));
+
```
[the remark this answers is not in the document]
Comment (swesam) [reply tests/integration.test.ts]: Replaced the fixed 2s sleep with a retry/poll loop (500ms intervals, max 5s). Faster when the API is responsive, more reliable when slow. Fixed in fb72a0d.
